app/models/peer_review.rb

```diff
@@ -8,9 +10,23 @@ class PeerReview < ActiveRecord::Base
   validates_presence_of :result
   validates_numericality_of :reviewer_id, only_integer: true, greater_than: 0
   validates_numericality_of :result_id, only_integer: true, greater_than: 0
+  validate :no_students_should_be_reviewer_and_reviewee
 
   def reviewee
     # TODO - Research optimizing or see if rails can do better
     Grouping.joins({ submissions: { results: :peer_reviews }}).where('peer_reviews.id = ?', self.id).first
   end
+
+  def no_students_should_be_reviewer_and_reviewee
+    if result and reviewer
```

Comment:
After further investigation, it appears that RSpec _always_ calls them with an object that has all nil fields, and then your proper object. The other tests in MarkUs do this, so I assume it's some kind of really weird standard. The factory has to be working right because it gets back a valid object for 1 of the 2 calls always.